### PR TITLE
Privacy dialog for organization

### LINKED ISSUE #1745: Implement privacy dialog
- [x] Possibility to change visualization privacy (dataset/map) #1872 
- [x] Show privacy options depending of the account (take into account upgrade links and open source).  #1872
- [x] Possibility to change permissions (R/RW) to users* [(screenshot)](https://cloud.githubusercontent.com/assets/978461/5859100/14951d12-a258-11e4-86d4-26b828825045.png)
- ~~Show affected users for datasets before revoking permissions* [(screenshot)](https://cloud.githubusercontent.com/assets/978461/5741236/30b02d24-9c0a-11e4-8245-6675e8681e2f.png)~~ still must-to-have but needs more thought, postponed #1963
- ~~Possibility to search/filter users to give permissions~~* nice-to-have ATM, postponed #1923

_*organization environment_

### PR DESCRIPTION
**Pull request by viddo**
fixes #1745 

solution to remaining issues (that were not postponed).

Also:
- Extracted `.Dialog-headerIcon` styles to `.LayoutIcon` as defined in style guide.
- Extracted common styles for the round navigational buttons used in dialogs, named `.NavButton`

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1295/))

**Comment by viddo** [comment]
there is a CSS bug with how the share-view is rendered which I'm working on, but meanwhile the code can be reviewed. ping @xavijam @javisantana

File: lib/assets/javascripts/cartodb/models/vis.js [hunk 1]
```diff
@@ -378,6 +378,18 @@ cdb.admin.Visualization = cdb.core.Model.extend({
     return _.map((this.permission.acl.toArray() || []), function(aclItem) {
       return aclItem.get('entity')
     });
+  },
+  
+  privacyOptions: function() {
+    var all = [ 'PUBLIC', 'LINK', 'PRIVATE', 'PASSWORD' ];
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/models/vis.js]: Shouldn't this variable be as a "static" var?

File: lib/assets/test/spec/cartodb/models/permissions.spec.js [hunk 2]
```diff
@@ -210,4 +210,82 @@ describe('cdb.admin.Permission', function () {
...
+    describe('given a model that has no access', function() {
+      it('should return false', function() {
+        model.removePermission(user1);
+        expect(model.canWrite(user1)).toBeFalsy();
+
+      });
+    });
+  });
+
 });
```
Comment (xavijam) [inline lib/assets/test/spec/cartodb/models/permissions.spec.js]: Yeah, permissions tests are important, :+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1297/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1299/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1300/))

**Comment by viddo** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1302/))

**Comment by viddo** [comment]
@xavijam CSS bug solved, let me know if this PR is ok

**Comment by xavijam** [comment]
:+1:

**Pull request merged**